calicoctl/commands/resources.go

```diff
@@ -124,6 +124,16 @@ func getResourceFromArguments(args map[string]interface{}) (unversioned.Resource
 			p.Metadata.CIDR = *cidr
 		}
 		return *p, nil
+	case "bgpPeer":
+		p := api.NewBGPPeer()
+		if name != "" {
+			err := p.Metadata.PeerIP.UnmarshalText([]byte(name))
+			if err != nil {
+				return nil, err
+			}
+		}
+		p.Metadata.Hostname = hostname
```

Comment:
Yes as per my TODO.

However, it is not as simple as that.  The IP index is used to construct an IP-version specific datastore key - therefore with the current data model it is necessary to have the IP address as an index field (i.e. have it in the Metadata).   I assume that was the reason you kept the CIDR as an ID field for the IP Pool rather than having an arbitrary name field?